agent/agent.go

```diff
@@ -394,15 +395,23 @@ func (a *Agent) write() {
 			if _, err := a.conn.Write(p); err != nil {
 				tracing.FinishSpan(data.ctx, err)
 				if data.typ == message.Response {
-					metrics.ReportTimingFromCtx(data.ctx, a.metricsReporters, handlerType, true)
+					metrics.ReportTimingFromCtx(data.ctx, a.metricsReporters, handlerType, err)
 				}
 				logger.Log.Error(err.Error())
 				return
 			}
 			var e error
 			tracing.FinishSpan(data.ctx, e)
 			if data.typ == message.Response {
-				metrics.ReportTimingFromCtx(data.ctx, a.metricsReporters, handlerType, m.Err)
+				var rErr error
+
+				if m.Err {
+					rErr = &errors.Error{}
+					// TODO: should I ignore this error?
```

Comment:
I intend to remove this comment after some discussion.

Reply:
What's the result if m.Data is not a valid error?

Reply:
That would be a mismatch between data.payload and data.err. Is it possible? But I could check if code is not empty to continue.

Reply:
Maybe set a default value just to be safe?